internal/extract.go

```diff
@@ -82,10 +82,9 @@ func extractOne(tarInterpreter TarInterpreter, source io.Reader) error {
 	return nil
 }
 
-// TODO : unit tests
 // Ensures that file extension is valid. Any subsequent behavior
 // depends on file type.
-func decryptAndDecompressTar(writer io.Writer, readerMaker ReaderMaker, crypter crypto.Crypter) error {
```

Comment:
Instead of extracting this method, change extract_test package to internal package.
It's "go way" to test in the same package

Reply:
This file imports testtools, and testtools import internal package. If I change the file package to internal, there will be a cyclic import. I am not sure that we could/should address that issue here, but instead it should be a part of a big task of changing all tests to go way.